trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/rest/RequestHeadersBuilder.java

```diff
@@ -0,0 +1,190 @@
+            "x-amzn-trace-id",
+            "expect",
+            "accept-encoding",
+            "user-agent",
+            "connection",
+            "amz-sdk-invocation-id",
+            "amx-sdk-request",
+            "x-amz-content-sha256",
+            "host");
+
+    record InternalRequestHeaders(
+            RequestHeaders requestHeaders,
+            Optional<RequestAuthorization> requestAuthorization,
+            Optional<Instant> requestDate,
+            Optional<Integer> contentLength,
+            Optional<Integer> decodedContentLength,
+            Optional<RequestContent.ContentType> requestPayloadContentType)
+    {
+        InternalRequestHeaders {
+            requireNonNull(requestHeaders, "requestHeaders is null");
+            requireNonNull(requestAuthorization, "requestAuthorization is null");
+            requireNonNull(requestDate, "requestDate is null");
+            requireNonNull(contentLength, "contentLength is null");
+            requireNonNull(decodedContentLength, "decodedContentLength is null");
+            requireNonNull(requestPayloadContentType, "requestPayloadContentType is null");
+        }
+    }
+
+    static InternalRequestHeaders parseHeaders(MultiMap allRequestHeaders)
+    {
+        Builder builder = new Builder();
+        allRequestHeaders.forEach((headerName, headerValues) -> {
+            switch (headerName) {
+                case "authorization", "x-amz-security-token" -> {} // these get handled separately
+                case "content-length" -> builder.contentLength(headerValues);
+                case "x-amz-decoded-content-length" -> builder.decodedContentLength(headerValues);
+                case "content-encoding" -> builder.contentEncoding(headerValues);
+                case "transfer-encoding" -> builder.transferEncoding(headerValues);
+                case "x-amz-date" -> builder.requestDate(headerValues);
+                default -> {
+                    if (!IGNORED_HEADERS.contains(headerName)) {
+                        builder.addPassthroughHeader(headerName, headerValues);
+                    }
+                }
+            }
+        });
+        allRequestHeaders.getFirst("authorization").map(authorization -> RequestAuthorization.parse(authorization, allRequestHeaders.getFirst("x-amz-security-token"))).ifPresent(builder::requestAuthorization);
+
+        return builder.build(allRequestHeaders);
+    }
+
+    private static class Builder
+    {
+        private final ImmutableMultiMap.Builder passthroughHeadersBuilder = ImmutableMultiMap.builder(false);
+        private Optional<RequestAuthorization> requestAuthorization = Optional.empty();
+        private Optional<Instant> requestDate = Optional.empty();
+        private Optional<Integer> contentLength = Optional.empty();
+        private Optional<Integer> decodedContentLength = Optional.empty();
+        private Optional<RequestContent.ContentType> requestPayloadContentType = Optional.empty();
```

Comment:
- Static import `RequestContent.ContentType`
- Suggestion: rename to `requestContentType`

Reply:
Fixed the static import
I wanted to avoid using `requestContentType` since this is not the content type of the request (which is what's indicated in the `content-type` header), but rather an indicator of the type/transfer encoding of the payload